.github/workflows/windows.yml

```
@@ -188,10 +189,73 @@ jobs:
           name: ${{ matrix.os }}-unit-reports
           path: ${{ env.GOPATH }}\src\github.com\docker\docker\bundles\*
 
+  unit-test-report:
+    runs-on: ubuntu-latest
+    if: always()
+    needs:
+      - unit-test
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - windows-2019
+          - windows-2022
+    steps:
+      -
+        name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      -
+        name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ matrix.os }}-unit-reports
+          path: /tmp/artifacts
+      -
+        name: Install teststat
+        run: |
+          go install github.com/vearutop/teststat@${{ env.TESTSTAT_VERSION }}
+      -
+        name: Create summary
+        run: |
+          teststat -markdown $(find /tmp/artifacts -type f -name '*.json' -print0 | xargs -0) >> $GITHUB_STEP_SUMMARY
+
+  integration-cli-prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.tests.outputs.matrix }}
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      -
+        name: Install gotestlist
+        run:
+          go install github.com/crazy-max/gotestlist/cmd/gotestlist@v0.2.0
+      -
+        name: Create matrix
+        id: tests
+        working-directory: ./integration-cli
+        run: |
+          # Distribute integration-cli tests for the matrix in integration-test job.
+          # Also prepend ./... to the matrix. This is a special case to run "Test integration" step exclusively.
+          echo ::set-output name=matrix::$(gotestlist -d 4 ./... | jq -c '. |= ["./..."] + .')
```

I would suggest splitting this so that if `gotestlist` fails for any reason, we get an explicit failure instead of just an empty matrix:

```suggestion
          matrix="$(gotestlist -d 4 ./...)"
          matrix="$(echo "$matrix" | jq -c '. |= ["./..."] + .')"
          echo "::set-output name=matrix::$matrix"
```

(due to `set -e` quirks :see_no_evil:)